src/indexable.cr

```diff
@@ -809,6 +809,35 @@ module Indexable(T)
     index(offset) { |e| yield e } || raise Enumerable::NotFoundError.new
   end
 
+  # Returns the first element in the indexable for which the passed block
+  # is truthy, starting from the given *offset*.
+  #
+  # Accepts an optional parameter *if_none*, to set what gets returned if
+  # no element is found (defaults to `nil`).
+  #
+  # ```
+  # [1, 2, 3, 4].find { |i| i > 2 }        # => 3
+  # [1, 2, 3, 4].find(-1, 2) { |i| i < 2 } # => -1
+  # [1, 2, 3, 4].find(-1) { |i| i > 8 }    # => -1
+  # ```
+  def find(offset : Int = 0, if_none = nil, & : T ->)
+    offset += size if offset < 0
+    return nil if offset < 0
+    return (index(offset) { |i| yield i }).try { |i| unsafe_fetch(i) } || if_none
+  end
+
+  # Returns the first element in the indexable for which the passed block is truthy.
+  # Raises `Enumerable::NotFoundError` if there is no element for which the block is truthy.
+  #
+  # ```
+  # [1, 2, 3, 4].find! { |i| i > 2 } # => 3
+  # [1, 2, 3, 4].find! { |i| i > 8 } # => raises Enumerable::NotFoundError
+  def find!(offset : Int = 0, & : T ->)
+    offset += size if offset < 0
+    return nil if offset < 0
+    return (index(offset) { |i| yield i }).try { |i| unsafe_fetch(i) } || raise Enumerable::NotFoundError.new
```

Comment:
Might make sense to follow the pattern of the other `!` bang methods:
```suggestion
    find(offset) || raise Enumerable::NotFoundError.new
```
Otherwise `find!` could return `nil` if given an invalid offset while the other methods like `index!` would raise.

Reply:
thanks, addressed!

Reply:
I'm not sure this is entirely addressed. If you do something like `collection.find! -1` that'll return `nil` as it stands. Which we likely don't want to happen for this bang version.

Reply:
If `find(offset)` returns nil, wouldn't the or statement lead to the raise being triggered?

Reply:
If I understand right, you're saying that if offset < 0, then we still have a possibility of returning `nil` right? I thought this behavior might be ok as today `index` and `index!` behave the same way within `Indexable`

Reply:
Thanks for giving the examples - this makes sense! I've updated the PR to remove the offset check in `find!`, as `find` would perform that anyway - and this way nil cannot be a possible return value in `find!`